src/components/TodoContainer/TodoContainer.jsx

```diff
@@ -0,0 +1,266 @@
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+};
+
+function TodoContainer() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchValue, setSearchValue] = useState("");
+  const [isAscending, setIsAscending] = useState(true);
+  const debouncedSearchValue = useDebounce(searchValue, 500);
+
+  const url = `https://api.airtable.com/v0/${
+    import.meta.env.VITE_AIRTABLE_BASE_ID
+  }/${import.meta.env.VITE_TABLE_NAME}`;
+
+  const fetchData = async (search = "") => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    };
+    try {
+    const response = await fetch(
+        `${url}?view=Grid%20view&sort[0][field]=createdTime&sort[0][direction]=${
+          isAscending ? "asc" : "desc"
+        }&filterByFormula=AND(SEARCH("${search}", {Title}), {completedAt} = "")`,
+        options
+      );
+      
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return {
+          id: todo.id,
+          title: todo.fields.Title,
+          createdTime: todo.fields.createdTime,
+          completedAt: !!todo.fields.completedAt,
+          dueDate: todo.fields.dueDate,
+        };
+      });
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [isAscending]);
+
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    }
+  }, [todoList, isLoading]);
+
+  useEffect(() => {
+    if (debouncedSearchValue) {
+      fetchData(debouncedSearchValue);
+    } else {
+      fetchData();
+    }
+  }, [debouncedSearchValue]);
+
+  const addTodo = async (newTodo) => {
+    const options = {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ fields: { Title: newTodo.title, dueDate: newTodo.dueDate } }),
+    };
+
+    try {
+      const response = await fetch(url, options);
+      console.log({ response });
+
+      if (!response.ok) {
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const addTodoData = await response.json();
+      const newTodo = {
+        id: addTodoData.id,
+        title: addTodoData.fields.Title,
+        createdTime: addTodoData.fields.createdTime,
+        completedAt: false,
+        dueDate: addTodoData.fields.dueDate,
+      };
+
+      const updatedTodoList = [...todoList, newTodo];
```

Comment:
Don't need to sort by created date here given this is adding a new todo (so should always be the latest)